### ISSUE TITLE
Modify compute API to support storage volumes

### ISSUE THREAD

**Issue by kaccardi** [01org/ciao#373]
Implement the Open Stack compute API changes required to support volume attachments.

**Comment by kaccardi** [comment]
Change existing create server apis to include support for the block-device-mapping-v2 parameter to boot from volume
Change existing list server apis to include support for listing attached volumes
Add support for os-volume_attachements apis (4 of them)

**Comment by kaccardi** [comment]
not done yet - working on moving compute API to new openstack package first.

**Comment by tpepper** [comment]
Marking as unassigned as it's blocked.  When unblocked whoever is free can take this on.

**Comment by kaccardi** [comment]
The block-device-mapping-v2 parameter is specified in the compute api spec as part of the create server api.
http://developer.openstack.org/api-ref/compute/?expanded=create-server-detail#create-server

**Comment by kaccardi** [comment]
this work item is unblocked now.

**Comment by tpepper** [comment]
Discussing with @kaccardi, the four os-volume_attachments APIs (POST, GET, GET {attachment_id}, DELETE) are duplicated versus APIs in Cinder which are already implemented elsewhere in Ciao, so these are probably lower priority / optional.

**Comment by tpepper** [comment]
With the os-volume_attachments APIs on the low priority list, that leaves:

* block-device-mapping-v2 implementation
* list servers implementation

as the priority work in this ticket

**Comment by tpepper** [comment]
As far as I can see, the list servers implementation is currently in order.  Which means the focus here is the block-device-mapping-v2.

**Comment by tpepper** [comment]
Addressed by https://github.com/01org/ciao/pull/896
